lib/Form/Listing/FormListingRepository.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace MailPoet\Form\Listing;
+
+use MailPoet\Entities\FormEntity;
+use MailPoet\Listing\ListingDefinition;
+use MailPoet\Listing\ListingRepository;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoetVendor\Doctrine\ORM\QueryBuilder;
+
+class FormListingRepository extends ListingRepository {
+  public function getGroups(ListingDefinition $definition): array {
+    $queryBuilder = clone $this->queryBuilder;
+    $this->applyFromClause($queryBuilder);
+    $this->applyParameters($queryBuilder, $definition->getParameters());
+
+    // total count
+    $countQueryBuilder = clone $queryBuilder;
+    $countQueryBuilder->select('COUNT(f) AS formCount');
+    $countQueryBuilder->andWhere('f.deletedAt IS NULL');
+    $totalCount = (int)$countQueryBuilder->getQuery()->getSingleScalarResult();
+
+    // trashed count
+    $trashedCountQueryBuilder = clone $queryBuilder;
+    $trashedCountQueryBuilder->select('COUNT(f) AS formCount');
+    $trashedCountQueryBuilder->andWhere('f.deletedAt IS NOT NULL');
+    $trashedCount = (int)$trashedCountQueryBuilder->getQuery()->getSingleScalarResult();
+
+    return [
+      [
+        'name' => 'all',
+        'label' => WPFunctions::get()->__('All', 'mailpoet'),
+        'count' => $totalCount,
+      ],
+      [
+        'name' => 'trash',
+        'label' => WPFunctions::get()->__('Trash', 'mailpoet'),
+        'count' => $trashedCount,
+      ],
+    ];
+  }
+
+  protected function applySelectClause(QueryBuilder $queryBuilder) {
+    $queryBuilder->select("PARTIAL f.{id,name,status,settings,createdAt,updatedAt,deletedAt}");
+  }
+
+  protected function applyFromClause(QueryBuilder $queryBuilder) {
+    $queryBuilder->from(FormEntity::class, 'f');
+  }
+
+  protected function applyGroup(QueryBuilder $queryBuilder, string $group) {
+    // include/exclude deleted
+    if ($group === 'trash') {
+      $queryBuilder->andWhere('f.deletedAt IS NOT NULL');
+    } else {
+      $queryBuilder->andWhere('f.deletedAt IS NULL');
+    }
+  }
+
+  protected function applySorting(QueryBuilder $queryBuilder, string $sortBy, string $sortOrder) {
+    $queryBuilder->addOrderBy("f.$sortBy", $sortOrder);
+  }
+
+  protected function applySearch(QueryBuilder $queryBuilder, string $search) {
```

Comment:
I'm unsure what is the best thing to do with this method and the other two below it. It seems to me that the page that lists the forms does not support search, filters, and parameters, but those three methods must exist as we are extending ListingRepository.

Reply:
Ahh looks like we kind of expected that all listings will have some filters, search, etc 🤔 I guess we could alter the ListingRepository and mark those methods not abstract or keep them here just updating those Todo comments to some comments saying that we don't support the functionality for the forms listing.